migrate/v0_15/migrate_test.go

```diff
@@ -99,3 +103,89 @@ func TestIncentive(t *testing.T) {
 
 	require.JSONEq(t, string(bz), string(appState[v0_15incentive.ModuleName]))
 }
+
+// Compare migration against auto-generated snapshot to catch regressions
+func TestAuth_Snapshot(t *testing.T) {
+	bz, err := ioutil.ReadFile(filepath.Join("testdata", "kava-7-test-auth-state.json"))
+	require.NoError(t, err)
+	appState := genutil.AppMap{auth.ModuleName: bz}
+
+	MigrateAppState(appState)
+
+	if _, err := os.Stat(filepath.Join("testdata", "kava-8-test-auth-state.json")); os.IsNotExist(err) {
+		err := ioutil.WriteFile(filepath.Join("testdata", "kava-8-test-auth-state.json"), appState[auth.ModuleName], 0644)
+		require.NoError(t, err)
+	}
+
+	snapshot, err := ioutil.ReadFile(filepath.Join("testdata", "kava-8-test-auth-state.json"))
+	require.NoError(t, err)
+
+	assert.JSONEq(t, string(snapshot), string(appState[auth.ModuleName]), "expected auth state snapshot to be equal")
+}
+
+func TestAuth_ParametersEqual(t *testing.T) {
+	bz, err := ioutil.ReadFile(filepath.Join("testdata", "kava-7-test-auth-state.json"))
+	require.NoError(t, err)
+
+	var genesisState auth.GenesisState
+	cdc := app.MakeCodec()
+	cdc.MustUnmarshalJSON(bz, &genesisState)
+
+	migratedGenesisState := Auth(genesisState, GenesisTime)
+
+	assert.Equal(t, genesisState.Params, migratedGenesisState.Params, "expected auth parameters to not change")
+}
+
+func TestAuth_AccountConversion(t *testing.T) {
+	bz, err := ioutil.ReadFile(filepath.Join("testdata", "kava-7-test-auth-state.json"))
```

Comment:
Mostly curiosity, but do you know at what block this was exported?

Reply:
No, looks like it was a current height export on 2021-08-05.  I can update to a specific height